### PR TITLE
Release 10.0.0 beta.1

### PR DESCRIPTION
**Pull request by davidyuk**

### CODE REVIEW

File: docs/guides/migration/10.0.0.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+### `contractCallStatic`, `contractCall`, `contractDeploy` are deprecated now ([c4ec019](https://github.com/aeternity/aepp-sdk-js/commit/c4ec019372f5d5745378e781afb1eb6b2b5acfeb))
+
+Use `getContractInstance` instead. Also, these methods will accept JavaScript-type variables instead of
+Sophia-encoded. For example:
+```js
+sdk.contractCallStatic(source, address, methodName, ['42'])
+```
+should be replaced with
+```js
+sdk.contractCallStatic(source, address, methodName, ['42'])
```
Comment (thepiwo) [inline docs/guides/migration/10.0.0.md:61]: is that supposed to be the same?

File: docs/guides/migration/10.0.0.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+### switched to @aeternity/aepp-calldata package ([#1313](https://github.com/aeternity/aepp-sdk-js/pull/1313))
+
+Numbers in Sophia are not limited in size. Before they were returned from contract methods as usual
+JavaScript numbers that have limited accuracy. To fix this, we make it return instances of BigInt.
+
+Variant types are now supported by sdk, so replace `"RelativeTTL(50)"` with `{ RelativeTTL: [50] }`.
+As an exception `Some(value)` is converted to the exact value in JavaScript and `None` is converted to `undefined`
+(Sophia's `option` type).
+
+Contract methods will accept/return instances of JavaScript's `Map` as variables of Sophia's `map`
```
Comment (thepiwo) [inline docs/guides/migration/10.0.0.md:86]: maybe we should mention that objects are not accepted any longer

File: docs/guides/migration/10.0.0.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Sophia's `hash`, `signature`, `bytes` types return values as `Uint8Array` instead of a hex-encoded
+string.
+
+Check the [documentation](https://github.com/aeternity/aepp-calldata-js/blob/master/README.md)
+of calldata package for additional info.
+
+Use `.decodedResult` instead of `.decode()` to get the result of method call.
+
+### pass source in options of `sdk.getContractInstance` ([5c690d2](https://github.com/aeternity/aepp-sdk-js/commit/5c690d2d77a433c8496ab9cda6fd66e52a6b4b23))
```
Comment (thepiwo) [inline docs/guides/migration/10.0.0.md:97]: maybe we can also add an example to pass aci?

File: docs/guides/migration/10.0.0.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+### additional options of `getContractInstance` accepted as usual ones ([10fb7ba](https://github.com/aeternity/aepp-sdk-js/commit/10fb7bad8f61b973c1be5daec50909d251ec1f90))
+
+For example, replace
+```js
+sdk.getContractInstance(source, { opt: { ttl: 1 } })
+```
+with
+```js
+sdk.getContractInstance(source, { ttl: 1 })
```
Comment (thepiwo) [inline docs/guides/migration/10.0.0.md:49]: is that still correct with the source as part of the object changes?

Comment (marc0olo) [reply docs/guides/migration/10.0.0.md:49]: I haven't checked the guide yet. do we have now also instructions how to deploy a contract only with bytecode and aci? should be possible now, right?

File: docs/guides/migration/10.0.0.md (hunk 1 above)
Comment (davidyuk) [reply docs/guides/migration/10.0.0.md]: nope, I forgot to replace string with number

File: docs/guides/migration/10.0.0.md (hunk 3 above)
Comment (davidyuk) [reply docs/guides/migration/10.0.0.md:86]: I intended to write the exactly migration guide, without mentioning new features, but it is ok to add this

File: docs/guides/migration/10.0.0.md (hunk 4 above)
Comment (davidyuk) [reply docs/guides/migration/10.0.0.md]: > is that still correct with the source as part of the object changes?

Nope, the changes are explained as they happen in history. I've fixed it by moving it below than putting `source` into options.

**Comment by davidyuk** [comment]
The issues of beta release I see:
- is not compatible with Webpack 4 https://github.com/aeternity/aepp-calldata-js/issues/91
- it depends on a branch in calldata repo, if that branch would be removed or rebased then npm won't be able to install this release of SDK

File: docs/guides/migration/10.0.0.md (hunk 3 above)
Comment (thepiwo) [reply docs/guides/migration/10.0.0.md:86]: also fine by me

**Comment by thepiwo** [comment]
are there any blockers for release?

**Comment by marc0olo** [comment]
> are there any blockers for release?

I think we should at least address this as it also contains breaking changes:
- https://github.com/aeternity/aepp-sdk-js/pull/1334

not sure about #1286 and #1332 right now. personally I am a bit concerned right now that we release a major version with breaking changes and possible another major version with breaking changes shortly after.

if we consider this beta to be published knowing that the full `v10.0.0` could have additional breaking changes I don't see any blocker at the moment.

opinions? @mradkov @davidyuk

**Comment by marc0olo** [comment]
the blockers that @davidyuk mentioned above are solved

**Comment by mradkov** [comment]
> if we consider this beta to be published knowing that the full v10.0.0 could have additional breaking changes I don't see any blocker at the moment.

Yes, lets do that so give enough time for people to experiment with `10.0.0-beta.1` maybe they found some issue which we can address in the official release as well.

**Review by thepiwo — APPROVED** [review]
lgtm

**Review by marc0olo — APPROVED** [review]

**Review by mradkov — APPROVED** [review]

**Pull request merged**